### ISSUE TITLE
TypeAbi markup should export the struct into .abi.json

### ISSUE THREAD

**Issue by razvanphp** [multiversx/mx-sdk-rs#1247]
We have some custom attributes defined using nested structs like:

```
#![no_std]

multiversx_sc::imports!();
multiversx_sc::derive_imports!();

use multiversx_sc_modules::{default_issue_callbacks, subscription};

#[derive(TypeAbi, TopEncode, TopDecode)]
pub struct FixedAttributes {
    pub boosts: u64,
    pub huds: u64,
    pub sounds: u64,
    pub max_rpm: u64,
}

#[derive(TypeAbi, NestedEncode, NestedDecode)]
pub struct Attributes<M: ManagedTypeApi> {
    pub boosts: u64,
    pub huds: u64,
    pub sounds: u64,
    pub races_won: u64,
    pub max_rpm: u64,
    pub battery: u64,
    pub license_type: ManagedBuffer<M>,
    pub renewable: bool,
    pub rechargeable: bool,
}

#[multiversx_sc::contract]
pub trait MxContractsRs: default_issue_callbacks::DefaultIssueCallbacksModule + subscription::SubscriptionModule {

[...]
```

we use the [subscription module](https://github.com/multiversx/mx-sdk-rs/blob/master/contracts/modules/src/subscription.rs#L27) to store the custom attributes on top of expiration and encode them in the `attributes` field of the NFT.

We want to have all the struct objects exported in the `output/<name>.abi.json` for displaying those attributes in the frontend in a reliable way.

Current workaround suggested by @andrei-marinica is to create a dummy getter function using the struct.

Thank you!
R

**Comment by andrei-marinica** [comment]
Fixed in PR https://github.com/multiversx/mx-sdk-rs/pull/1253

**Comment by andrei-marinica** [comment]
Released in v0.44.0.
